source/D2Common/src/Drlg/DrlgRoomTile.cpp

```diff
@@ -143,104 +127,103 @@ D2DrlgTileDataStrc* __fastcall DRLGROOMTILE_InitWallTileData(D2RoomExStrc* pRoom
 	pTileData->nBlue = -1;
 	pTileData->nRed = -1;
 
-	DRLGROOMTILE_InitializeTileDataFlags(pRoomEx, pTileData, nTileFlags, nTileType, nX, nY);
+	DRLGROOMTILE_InitializeTileDataFlags(pRoomEx, pTileData, nPackedTileInformation, nTileType, nX, nY);
 
 	if (nTileType == TILETYPE_RIGHTPARTOFNORTHCORNERWALL)
 	{
-		DRLGROOMTILE_InitializeTileDataFlags(pRoomEx, DRLGROOMTILE_InitWallTileData(pRoomEx, ppTileData, nX, nY, nTileFlags, DRLGROOMTILE_GetTileCache(pRoomEx, 4, nTileFlags), 4), nTileFlags, 4, nX, nY);
+		DRLGROOMTILE_InitializeTileDataFlags(pRoomEx, DRLGROOMTILE_InitWallTileData(pRoomEx, ppTileData, nX, nY, nPackedTileInformation, DRLGROOMTILE_GetTileCache(pRoomEx, TILETYPE_LEFTPARTOFNORTHCORNERWALL, nPackedTileInformation), TILETYPE_LEFTPARTOFNORTHCORNERWALL), nPackedTileInformation, TILETYPE_LEFTPARTOFNORTHCORNERWALL, nX, nY);
 	}
 
 	return pTileData;
 }
 
 //D2Common.0x6FD88AC0
-void __fastcall DRLGROOMTILE_InitializeTileDataFlags(D2RoomExStrc* pRoomEx, D2DrlgTileDataStrc* pTileData, int nTileFlags, int nType, int nX, int nY)
+void __fastcall DRLGROOMTILE_InitializeTileDataFlags(D2RoomExStrc* pRoomEx, D2DrlgTileDataStrc* pTileData, uint32_t nTileFlags, int nType, int nX, int nY)
 {
-	int nFlags = 0;
+	const D2C_PackedTileInformation nTileInfo{ nTileFlags };
 
-	if (pRoomEx && (nType == 9 || nType == 8))
+	if (pRoomEx && (nType == TILETYPE_RIGHTWALLWITHDOOR || nType == TILETYPE_LEFTWALLWITHDOOR))
 	{
 		DRLGROOMTILE_AddTilePresetUnits(pRoomEx, pTileData, nTileFlags, nX, nY, 0);
 	}
 
-	if (nType != 13)
+	if (nType != TILETYPE_SHADOWS)
 	{
-		pTileData->dwFlags |= ((((unsigned int)nTileFlags >> 18) & 3) + 1) << 14;
+		pTileData->dwFlags |= (nTileInfo.nWallLayer + 1) << MAPTILE_WALL_LAYER_BIT;
 	}
 
-	if (nType == 14)
+	if (nType == TILETYPE_TREES)
 	{
-		pTileData->dwFlags |= 4;
+		pTileData->dwFlags |= MAPTILE_TREES;
 	}
-	else if (nType == 11 || nType == 10 || nType == 9 || nType == 8)
+	else if (nType == TILETYPE_SPECIALTILES_11 || nType == TILETYPE_SPECIALTILES_10 || nType == TILETYPE_RIGHTWALLWITHDOOR || nType == TILETYPE_LEFTWALLWITHDOOR)
 	{
-		pTileData->dwFlags |= 2;
+		pTileData->dwFlags |= MAPTILE_WALL_EXIT;
 	}
 
-	if (nTileFlags & 0x80)
+	if (nTileInfo.bLayerAbove)
 	{
-		pTileData->dwFlags |= 1;
+		pTileData->dwFlags |= MAPTILE_UNK_0x1;
 	}
 
-	if (nTileFlags & 0x10000000)
+	if (nTileInfo.bLinkage)
 	{
-		pTileData->dwFlags |= 0x102;
+		pTileData->dwFlags |= MAPTILE_FLOOR_LINKER_PATH | MAPTILE_WALL_EXIT;
 	}
 
-	if (nTileFlags & 0x20000)
+	if (nTileInfo.bUnwalkable)
 	{
-		pTileData->dwFlags |= 0x40;
+		pTileData->dwFlags |= MAPTILE_UNWALKABLE;
 	}
 
-	if (nTileFlags & 0x10000)
+	if (nTileInfo.bFillLOS)
 	{
-		pTileData->dwFlags |= 0x80;
+		pTileData->dwFlags |= MAPTILE_FILL_LOS;
 	}
 
-	if (nTileFlags & 8)
+	if (nTileInfo.bEnclosed)
 	{
-		pTileData->dwFlags |= 4;
+		pTileData->dwFlags |= MAPTILE_TREES;
 	}
 
-	if (nTileFlags >= 0)
+	if (nTileInfo.bHidden)
```

Comment:
This looks wrong to me. `nTileFlags >= 0` means `!(nTileFlags & 0x80000000)`, i.e. `!nTileInfo.bHidden` which also makes more sense with respect to the actions taken below.

Reply:
Nice catch, thanks